### PR TITLE
Feature/bg 14

### PR DESCRIPTION
**Pull request by JustD1d1t**
Add test for gamtimer:
1. gameTimerTextBegin - test time that will be displayed when game is starting.
2. gameTimerTextFinished - test time that should be displayed when time is over. It should be 00:00
3. gameTimerTextStop - test time that should be displayed after X seconds. It should be time at the beginning - X seconds

Refactor code for game timer funcitons and modal functions that will allow test simple functions

Add global styles to howtoplay page

Change classnames in modal instead styling using tags

### CODE REVIEW

File: src/game/gameTimer.js [hunk 1]
```diff
@@ -1,11 +1,8 @@
-import { finishGame } from "./gameResult";
-
 // container class where will be displayed timer. Change classname if it is necessary
-const TIMER = document.querySelector(".timer");
-
 let gameTime;
 
-export function setTimer(time = 10) {
+function setTimer(time = 10, cb) {
+  const TIMER = document.querySelector(".timer");
```
Comment (ruljin) [inline src/game/gameTimer.js]: `SCREAMING_SNAKE_CASE` is used to make constants more verbose and visible for a reader. If you are creating const in a function it doesn't have then any real value and it can be called just a `timer`.

File: src/game/gameTimer.js [hunk 2]
```diff
@@ -1,11 +1,8 @@
-import { finishGame } from "./gameResult";
-
 // container class where will be displayed timer. Change classname if it is necessary
-const TIMER = document.querySelector(".timer");
-
 let gameTime;
 
-export function setTimer(time = 10) {
+function setTimer(time = 10, cb) {
```
Comment (ruljin) [inline src/game/gameTimer.js]: In the case of `callback` - `cb` s overall good enough, but for fire try to choose more verbose names for passed variables that make it obvious for everyone and always, what It is.

File: src/game/test/gameTimerTextBegin.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @jest-environment jsdom
```
Comment (ruljin) [inline src/game/test/gameTimerTextBegin.test.js]: This comment is required for `jsdom` to work?

File: src/game/test/gameTimerTextBegin.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+"use strict";
+const { jest } = require("@jest/globals");
+const { setTimer } = require("../gameTimer");
+
+const timer = document.createElement("div");
+timer.classList.add("timer");
+document.querySelector("body").append(timer);
+
+const mockFunciton = jest.fn();
```
Comment (ruljin) [inline src/game/test/gameTimerTextBegin.test.js]: `function` 😄

File: src/game/test/gameTimerTextStop.test.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+const mockFunction = jest.fn();
+
+describe("Using function gameTimer and stopTimer", () => {
+  describe("parameteres provided", () => {
+    test.each`
+      input
+      ${{ minutes: undefined, textContent: "09:50", secondsToStop: 10 }}
+    `(
+      "returns $input.textContent when call stopTimer function after $input.secondsToStop seconds when first parameter for gameTimer funciton is $input.minutes",
```
Comment (ruljin) [inline src/game/test/gameTimerTextStop.test.js]: `function` 😄

**Review by ruljin — CHANGES_REQUESTED** [review]

File: src/game/gameTimer.js (hunk 1 above)
Comment (JustD1d1t) [reply src/game/gameTimer.js]: changed

File: src/game/gameTimer.js (hunk 2 above)
Comment (JustD1d1t) [reply src/game/gameTimer.js]: called `finishGame`

File: src/game/test/gameTimerTextBegin.test.js (hunk 3 above)
Comment (JustD1d1t) [reply src/game/test/gameTimerTextBegin.test.js]: yeah. It it necessary

File: src/game/test/gameTimerTextBegin.test.js (hunk 4 above)
Comment (JustD1d1t) [reply src/game/test/gameTimerTextBegin.test.js]: fixed :D i have to always look at this `funciton`

File: src/game/gameTimer.js [hunk 6]
```diff
@@ -1,11 +1,8 @@
-import { finishGame } from "./gameResult";
-
 // container class where will be displayed timer. Change classname if it is necessary
-const TIMER = document.querySelector(".timer");
-
 let gameTime;
 
-export function setTimer(time = 10) {
+function setTimer(time = 10, finishGame) {
```
Comment (ruljin) [inline src/game/gameTimer.js:4]: What I had on my mind was that using a full name would be enough. In shortcut, not `cb` but `callback` and info for the future that when passing, try to name things with their full name and give them names that are verbose enough.

IMO `callback` and `finishGame` both are good in this case.

Comment (JustD1d1t) [reply src/game/gameTimer.js:4]: if both cases are good i would like to stay with finisGame now

File: src/game/test/gameTimerTextBegin.test.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+timer.classList.add("timer");
+document.querySelector("body").append(timer);
+
+const mockFunction = jest.fn();
+
+describe("Using function setTimer", () => {
+  describe("parameteres provided", () => {
+    test.each`
+      input
+      ${{ minutes: 2, textContent: "02:00" }}
```
Comment (ruljin) [inline src/game/test/gameTimerTextBegin.test.js]: Make 3 or 5 examples to test. What is the point of testing only one variant in which everything gonna is ok? It should work for any time, yup? What will happen if I will pas 128? 2048? 0.5? Test not only obvious but also corner cases.

File: src/game/test/gameTimerTextFinished.test.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+jest.useFakeTimers();
+jest.spyOn(global, "setTimeout");
+
+const mockFunction = jest.fn();
+
+describe("Using function gameTime", () => {
+  describe("with provided parameters", () => {
+    test.each`
+      input
+      ${{ minutes: undefined, textContent: "00:00", secondsToStop: 10 * 60 }}
```
Comment (ruljin) [inline src/game/test/gameTimerTextFinished.test.js]: ☝️

File: src/game/test/gameTimerTextStop.test.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+jest.useFakeTimers();
+jest.spyOn(global, "setTimeout");
+
+const mockFunction = jest.fn();
+
+describe("Using function gameTimer and stopTimer", () => {
+  describe("parameteres provided", () => {
+    test.each`
+      input
+      ${{ minutes: undefined, textContent: "09:50", secondsToStop: 10 }}
```
Comment (ruljin) [inline src/game/test/gameTimerTextStop.test.js]: ☝️

**Review by ruljin — CHANGES_REQUESTED** [review]

File: src/game/test/gameTimerTextBegin.test.js (hunk 7 above)
Comment (JustD1d1t) [reply src/game/test/gameTimerTextBegin.test.js]: Add more scenarios and test for number smaller than 1, bigger than 60 and inputs in other format

File: src/game/test/gameTimerTextFinished.test.js (hunk 8 above)
Comment (JustD1d1t) [reply src/game/test/gameTimerTextFinished.test.js]: Add more scenarios

File: src/game/test/gameTimerTextStop.test.js (hunk 9 above)
Comment (JustD1d1t) [reply src/game/test/gameTimerTextStop.test.js]: add more scenarios

File: src/game/test/gameTimerText.test.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    test.each`
+      input
+      ${{ minutes: 2, textContent: "02:00" }}
+      ${{ minutes: undefined, textContent: "10:00" }}
+    `(
+      "returns $input.textContent when first parameter is $input.minutes at the beginning of game",
+      ({ input }) => {
+        try {
+          setTimer(input.minutes, mockFunction);
+        } catch (err) {}
```
Comment (ruljin) [inline src/game/test/gameTimerText.test.js]: Ok, there is a case that does ... nothing?

**Review by ruljin — APPROVED** [review]
